lib/imagereader/imdimagereader.cc

```diff
@@ -85,204 +86,196 @@ static unsigned getSectorSize(uint8_t flags)
+                        break;
+
+                    case 1: /* Normal data: (Sector Size) bytes follow */
+                        sectordata = br.read(sectorSize);
+                        headerPtr += sectorSize;
+                        sector->data.writer().append(sectordata);
+                        break;
+
+                    case 2: /* Compressed: All bytes in sector have same value (xx) */
+                        sectordata = br.read(1);
+                        headerPtr++;
+                        sector->data.writer().append(sectordata);
+
+                        for (int k = 1; k < sectorSize; k++)
+                        {
+                            //fill data till sector is full
+                            sector->data.writer().append(sectordata);
+                        }
+                        break;
+
+                    case 3: /* Normal data with "Deleted-Data address mark" */
+                    case 4: /* Compressed with "Deleted-Data address mark"*/
+                    case 5: /* Normal data read with data error */
+                    case 6: /* Compressed read with data error" */
+                    case 7: /* Deleted data read with data error" */
+                    case 8: /* Compressed, Deleted read with data error" */
+                    default:
+                        Error() << fmt::format("don't understand IMD disks with sector status {}", Status_Sector);
+                }       
+                sector->status = Sector::OK;
+                sector->logicalTrack = sector->physicalCylinder = header.track;
+                sector->logicalSide = sector->physicalHead = header.Head;
+                sector->logicalSector = (sector_map[s]);
+            }
+
+        }
+        //Write format detected in IMD image to screen to help user set the right write parameters
+
+        image->setGeometry({
+            .numTracks = header.track,
+            .numSides = header.Head + 1U,
+            .numSectors = header.numSectors,
+            .sectorSize = sectorSize
+        });
+
+        size_t headSize = header.numSectors * sectorSize;
         size_t trackSize = headSize * (header.Head + 1);
 
-		std::cout 	<< "reading IMD image\n"
-					<< fmt::format("{} tracks, {} heads; {}; {} kbps; {} sectoren; sectorsize {}; sectormap {}; {} kB total \n",
-					header.track, header.Head + 1,
-					mfm ? "MFM" : "FM",
-					Modulation_Speed, header.numSectors, sectorSize, sector_skew, (header.track+1) * trackSize / 1024);
+        std::cout << fmt::format("IMD: {} tracks, {} heads; {}; {} kbps; {} sectors; sectorsize {};\n"
+                                 "     sectormap {}; {} kB total\n",
+                    header.track, header.Head + 1,
+                    mfm ? "MFM" : "FM",
+                    Modulation_Speed, header.numSectors, sectorSize, sector_skew, (header.track+1) * trackSize / 1024);
+
+        if (config.encoder().format_case() == EncoderProto::FormatCase::FORMAT_NOT_SET)
```

Comment:
Convention wise I think it is OK for the specified profile to override the format's internal profile for logical image formats like img/fdi/dim, but I think for formats like imd and d88 that have explicit sector encodings, that should always override the profile.